riscv/execute.cc

```diff
@@ -261,7 +261,7 @@ void processor_t::step(size_t n)
             state.single_step = state.STEP_STEPPED;
           }
 
-          insn_fetch_t fetch = mmu->load_insn(pc);
+          insn_fetch_t fetch = _mmu->access_icache(pc)->data;
```

Comment:
I think this breaks if you enable/disable logs partway through execution, since the icache would end up containing a mix of `fast`/`logged` funcs.

Reply:
Unrelated, perhaps: What the point of `_mmu` here?

Reply:
I realize it's a risky functional change for other reasons, so let's just drop it.

Reply:
Reading `_mmu` at the top of this function allows it to be allocated in a callee-saved register, rather than re-reading it every time the compiler can't prove it hasn't been modified.  It's just one load, but it makes a slight difference.